emData/generate_MC.py

```diff
@@ -0,0 +1,185 @@
+
+# First, parse the wires file and store the memory names associated with MCs in
+# dictionaries with the MC names as keys.
+wiresFile = open(wiresFileName)
+CMMems = {}
+FMMems = {}
+for line in wiresFile:
+    # Only barrel-only seeds are supported right now.
+    if "MC_D" in line: continue # No disks for now
+    line = line.rstrip()
+    mcName = re.sub(r".*MC_(......).*", r"MC_\1", line)
+    memName = line.split()[0]
+    if memName.startswith("CM_"):
+        if mcName not in CMMems:
+            CMMems[mcName] = []
+        CMMems[mcName].append(memName)
+    if memName.startswith("FM_"):
+        FM = re.sub(r"FM_(....)_......", r"\1", memName)
+        if mcName not in FMMems:
+            FMMems[mcName] = []
+        FMMems[mcName].append(FM)
+wiresFile.close()
+
+# Open and print out preambles for the parameters and top files.
+dirname = os.path.dirname(os.path.realpath('__file__'))
+parametersFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculator_parameters.h"), "w")
+parametersFile.write(
+    "#ifndef TrackletAlgorithm_MatchCalculator_parameters_h\n"
+    "#define TrackletAlgorithm_MatchCalculator_parameters_h\n"
+    "\n"
+    "// This file contains numbers of memories and bit masks that are specific to\n"
+    "// each MatchCalculator and that come directly from the wiring.\n"
+    "//\n"
+    "// The validity of each of the barrel CM memories is determined by\n"
+    "// FMMask. The bits of this mask, from least significant to most\n"
+    "// significant, represent the memories in the order they are passed to\n"
+    "// MatchCalculator; e.g., the LSB corresponds to\n"
+    "// TF::L1L2. If a bit is set, the corresponding memory is\n"
+    "// valid, if it is not, the corresponding memory is not valid.\n"
+)
+topHeaderFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculatorTop.h"), "w")
+topHeaderFile.write(
+    "#ifndef TrackletAlgorithm_MatchCalculatorTop_h\n"
+    "#define TrackletAlgorithm_MatchCalculatorTop_h\n"
+    "\n"
+    "#include \"MatchCalculator.h\"\n"
+    "\n"
+)
+topFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculatorTop.cc"), "w")
+topFile.write(
+    "#include \"MatchCalculatorTop.h\"\n"
+    "\n"
+    "////////////////////////////////////////////////////////////////////////////////\n"
+    "// Top functions for various MatchCalculators (MC). For each iteration of\n"
+    "// the main processing loop, a MC retrieves an array of Candidate Matches,\n"
+    "// and computes Full Matches from each.\n"
+    "////////////////////////////////////////////////////////////////////////////////\n"
+)
+
+# Calculate parameters and print out parameters and top function for each MC.
+for mcName in sorted(CMMems.keys()):
+    seed = re.sub(r"MC_(..)....", r"\1", mcName)
+    iMC = re.sub(r"MC_.....(.)", r"\1", mcName)
+
+    # numbers of memories
+    nCMMem = len(CMMems[mcName])
+    nFMMem = 8#FIXME after TBHelper is added to the test bench len(FMMems[tpName])
```

Comment:
@aehart is this ok? I basically made this PR hang on #179. If this one is merged first, I can just push this change to the #179, otherwise I can push it here.